### ISSUE TITLE
Learning rate finder is not working with lazy parameters

### ISSUE THREAD

**Issue by GuillaumeRochette** [PyTorchLightning/pytorch-lightning#11737]
## 🐛 Bug

One cannot use the learning rate finder, aka `auto_lr_find`, if the model has any `UninitializedParameter`.
The error arises when the model state is restored after the search.

### To Reproduce

https://colab.research.google.com/drive/1L64i9svqfkmXYktvQCZpUAcGaJRIHDSB?usp=sharing

### Expected behavior

The model state should be saved after initialization and not before.

### Environment

* CUDA:
	- GPU:
		- Tesla K80
	- available:         True
	- version:           11.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.10.0+cu111
	- pytorch-lightning: 1.5.9
	- tqdm:              4.62.3
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.12
	- version:           #1 SMP Tue Dec 7 09:58:10 PST 2021

**Comment by 34j** [comment]
It seems that this can be forcibly solved by `patch()` and `deepcopy()`.
```python
from unittest.mock import patch

model_copy = deepcopy(model.cpu())
with patch.object(model, "load_state_dict", lambda _: None):
    lr_finder = tuner.lr_find(model, data_module, num_training=20)
model = model_copy
model.lr = lr_finder.suggestion()
```
